src/precice/bindings/python_future/precice_future.pyx

```diff
@@ -98,134 +98,140 @@ cdef class Interface:
         return self.thisptr.hasMesh (convert(mesh_name))
 
     # get mesh ID
-    def get_mesh_id (self, str mesh_name):
+    def get_mesh_id (self, mesh_name):
         return self.thisptr.getMeshID (convert(mesh_name))
 
     # get mesh IDs
     def get_mesh_ids (self):
         return self.thisptr.getMeshIDs ()
 
     # returns a handle to a created mesh
-    def get_mesh_handle(self, str mesh_name):
+    def get_mesh_handle(self, mesh_name):
         raise Exception("The API method get_mesh_handle is not yet available for the Python bindings.")
 
     # creates a mesh vertex
-    def set_mesh_vertex(self, int mesh_id, position):
+    def set_mesh_vertex(self, mesh_id, position):
+        dimensions = position.size
+        assert(dimensions == self.get_dimensions())
         cdef np.ndarray[double, ndim=1] _position = np.ascontiguousarray(position, dtype=np.double)
         vertex_id = self.thisptr.setMeshVertex(mesh_id, &_position[0])
         return vertex_id
 
     # returns the number of vertices of a mesh
-    def get_mesh_vertex_size (self, int mesh_id):
+    def get_mesh_vertex_size (self, mesh_id):
         return self.thisptr.getMeshVertexSize(mesh_id)
 
     # creates multiple mesh vertices
-    def set_mesh_vertices (self, int mesh_id, positions):
-        cdef np.ndarray[double, ndim=1] _positions = np.ascontiguousarray(positions, dtype=np.double)
-        size = _positions.size/self.get_dimensions()
-        assert(size.is_integer())
-        size = int(size)
+    def set_mesh_vertices (self, mesh_id, positions):
+        size, dimensions = positions.shape
+        assert(dimensions == self.get_dimensions())
+        cdef np.ndarray[double, ndim=1] _positions = np.ascontiguousarray(positions.flatten(), dtype=np.double)
         cdef np.ndarray[int, ndim=1] _ids = np.empty(size, dtype=np.int32)
         self.thisptr.setMeshVertices (mesh_id, size, &_positions[0], &_ids[0])
         return _ids
 
     # get vertex positions for multiple vertex ids from a given mesh
-    def get_mesh_vertices(self, int mesh_id, ids):
+    def get_mesh_vertices(self, mesh_id, ids):
         cdef np.ndarray[int, ndim=1] _ids = np.ascontiguousarray(ids, dtype=np.int32)
         size = _ids.size
         cdef np.ndarray[double, ndim=1] _positions = np.empty(size * self.get_dimensions(), dtype=np.double)
         self.thisptr.getMeshVertices (mesh_id, size, &_ids[0], &_positions[0])
-        return _positions
+        return _positions.reshape((size, self.get_dimensions()))
 
     # gets mesh vertex IDs from positions
-    def get_mesh_vertex_ids_from_positions (self, int mesh_id, positions):
-        cdef np.ndarray[double, ndim=1] _positions = np.ascontiguousarray(positions, dtype=np.double)
-        size = _positions.size/self.get_dimensions()
-        assert(size.is_integer())
-        size = int(size)
+    def get_mesh_vertex_ids_from_positions (self, mesh_id, positions):
+        size, dimensions = positions.shape
```

Comment:
Should be able to handle tuples and lists

Reply:
done.